ironfish/src/rpc/routes/accounts/getTransactions.ts

```diff
@@ -66,12 +69,41 @@ router.register<typeof GetAccountTransactionsRequestSchema, GetAccountTransactio
 
     const headSequence = await node.wallet.getAccountHeadSequence(account)
 
+    let capacity = -1
+    let queue = undefined
+    if (request.data.show) {
+      capacity = request.data.show
+      queue = new FastPriorityQueue<TransactionValue>(function (a, b) {
+        return a.transaction.expirationSequence() < b.transaction.expirationSequence()
+      })
+    }
+
     for await (const transaction of account.getTransactions()) {
-      if (request.closed) {
-        break
+      if (capacity !== -1) {
+        Assert.isNotUndefined(queue)
+        queue.add(transaction)
+        if (queue.size > capacity) {
+          queue.poll()
+        }
+      } else {
+        if (request.closed) {
+          break
+        }
+
+        await streamTransaction(request, node, account, transaction, { headSequence })
```

Comment:
The logic here is getting a bit confusing because we're supporting three different ways of fetching transactions.

It might be time to split this up into separate functions for the different use cases of this RPC route